units/crons.go

```diff
@@ -828,6 +829,14 @@ func PopulateHostCreationJobs(env evergreen.Environment, part int) amboy.QueueOp
 		catcher := grip.NewBasicCatcher()
 		submitted := 0
 
+		// shuffle hosts because hosts will always come off the index in insertion order
+		// and we want all of them to get an equal chance at being created on this pass
```

Comment:
nit: Could you add to the comment that this is the Fisher-Yates shuffle? I read it twice and almost convinced myself it was wrong, but then read the wikipedia article.